### PR TITLE
fix(actor): release popped passivation heap entries

### PR DESCRIPTION
**Pull request by notrodans**
## Summary

Clear the removed `passivationEntry` slot in `passivationHeap.Pop` before shrinking the slice.

Without clearing the slot, the heap's backing array keeps a strong reference to every popped `*passivationEntry`. Each entry keeps its `target` alive, and for grain passivation that target is a `*grainPID`. As a result, successfully deactivated grains can remain reachable through the passivation heap backing array even after they have been removed from the grain registry.

The fix is one line:

```go
old[n-1] = nil
```

This PR also adds a deterministic regression test that keeps a view of the original backing array and verifies that the popped slot is `nil` after `container/heap.Pop`.

## Retention chain

```text
actorSystem
  -> passivationManager
    -> passivationHeap backing array
      -> *passivationEntry
        -> target (passivationParticipant)
          -> *grainPID
            -> grain
            -> identity
            -> mailboxes
            -> config
            -> timers
            -> dependencies
```

Shrinking a slice only changes its length. It does not clear pointer values stored beyond the new length, so those objects remain reachable while the backing array itself remains reachable.

## Reproducer

The issue was found while stress-testing virtual grain passivation with 50,000 simultaneously active grains.

### Before the fix

```text
logical actors: 1000000
physical activations: 50000
goroutines: 20 -> 20 (delta 0)
active heap: 78.54 MB / 1647.02 bytes per activation
active heap+stacks: 78.57 MB / 1647.67 bytes per activation
allocated during activation: 516.91 MB / 10840.35 bytes per activation

after passivation:
retained heap over baseline: 78.24 MB
cleanup duration: 122.825863ms
```

The grain registry was empty after passivation, but pprof still showed the activation-owned structures retained, including `newGrainPID`, `newGrainConfig`, `newGrainTimers`, `newGrainIdentity`, and grain mailboxes.

### After the fix

```text
logical actors: 1000000
physical activations: 50000
goroutines: 20 -> 20 (delta 0)
active heap: 78.67 MB / 1649.75 bytes per activation
active heap+stacks: 78.73 MB / 1651.06 bytes per activation
allocated during activation: 516.92 MB / 10840.53 bytes per activation

after passivation:
retained heap over baseline: 3.06 MB
cleanup duration: 133.543734ms
```

The fix reduces retained heap after passivating the 50,000 grains from about 78 MB to about 3 MB, while keeping passivation latency in the same range.

### pprof before fix

The retained profile was dominated by grain-owned allocations such as:

```text
~25.5 MB  actor.newGrainMailbox
~21.5 MB  actor.newGrainPID
 ~7.5 MB  internal/xsync.NewMap[...]
 ~5.5 MB  actor.newGrainTimers
 ~4.5 MB  actor.newGrainConfig
 ~4.5 MB  actor.newGrainIdentity
```

### pprof after fix

After passivation with the fix, those grain-owned allocations no longer dominate the retained heap. The remaining memory is primarily runtime/shared-pool overhead rather than the 50,000 deactivated grain instances.

## Why the unit test does not rely on GC

A GC/finalizer regression test would be timing-sensitive. Instead, the test directly checks the invariant responsible for the retention:

```go
entry := &passivationEntry{id: "retained"}
queue := passivationHeap{entry}
backing := queue[:cap(queue)]

popped := heap.Pop(&queue)

require.Same(t, entry, popped)
require.Empty(t, queue)
require.Nil(t, backing[0])
```

This fails against the current implementation and passes with the fix.

## Scope

This does not change passivation behavior or scheduling semantics. It only removes an obsolete pointer from the backing array so deactivated participants can be reclaimed by the GC.

The large-scale reproducer and heap profiles came from an external GoAkt grains playground used to compare virtual-actor runtimes; the PR itself intentionally keeps only the minimal deterministic regression test.

### CODE REVIEW

File: actor/passivation_heap_retention_test.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package actor
+
+import (
+	cheaps "container/heap"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPassivationHeapPopClearsBackingArrayReference(t *testing.T) {
```
Comment (Tochemey) [inline actor/passivation_heap_retention_test.go]: I prefer test file go to their implementation file.

**Review by Tochemey — CHANGES_REQUESTED** [review]

**Review by Tochemey — APPROVED** [review]

**Comment by Tochemey** [comment]
@notrodans Thanks for this PR. This was an amazing fix
